### PR TITLE
Add a TextResponse to easily return plain text HTTP responses

### LINKED ISSUE #76: Add a TextResponse?
There is currently the base `Response`, but also `HtmlResponse`, `JsonResponse`, etc.

That's practical when we want to return HTML or JSON, but there is no simple `TextResponse` (or `PlainTextResponse`. That's not a huge deal, but writing a simple *hello world* is just plain confusing.

Example:

```php
// Super verbose
$response = new Response();
$res = $res->withHeader('content-type', 'text/plain');
$response->getBody()->write('Hello world!');
return $response;

// Wrong content type, but at least it's simpler and shorter
return new HtmlResponse('Hello world');
```

Do you think it would make sense to add a new `TextResponse` class?

Or even simpler if we could use the base Response class to return plain text content, but the $body parameter in the constructor seems to only accept streams. Maybe it could accept a plain string? (but then how to differentiate from a stream…)

### PR DESCRIPTION
**Pull request by mnapoli**
Fixes #76

I did not add a charset as the HtmlResponse didn't have one. Maybe this is something to add? (see e.g. [in symfony](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/HttpFoundation/Response.php#L282-L288)). I am very far from an expert in HTTP stuff, so I'm just bringing it up but I have no idea (and it should probably be done in a separate PR anyway).

### CODE REVIEW

File: doc/book/custom-responses.md [hunk 1]
```diff
@@ -23,6 +23,21 @@ Some standard use cases, however, make this un-wieldy:
 Starting with version 1.1, Diactoros offers several custom response types for simplifying these
 common tasks.
 
+## Text Responses
+
+`Zend\Diactoros\Response\TextResponse` allows creating a plain text response. It sets the
+`Content-Type` header to `text/html` by default:
```
Comment (akrabat) [inline doc/book/custom-responses.md]: text/plain, not text/html.

File: doc/book/custom-responses.md [hunk 2]
```diff
@@ -23,6 +23,21 @@ Some standard use cases, however, make this un-wieldy:
 Starting with version 1.1, Diactoros offers several custom response types for simplifying these
 common tasks.
 
+## Text Responses
+
+`Zend\Diactoros\Response\TextResponse` allows creating a plain text response. It sets the
```
Comment (akrabat) [inline doc/book/custom-responses.md]: replace "allows creating" with "creates".

File: src/Response/TextResponse.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+ * Allows creating a response by passing a string to the constructor;
+ * by default, sets a status code of 200 and sets the Content-Type header to
+ * text/plain.
+ */
+class TextResponse extends Response
+{
+    use InjectContentTypeTrait;
+
+    /**
+     * Create an HTML response.
```
Comment (akrabat) [inline src/Response/TextResponse.php]: Not HTML…

File: src/Response/TextResponse.php [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+     *
+     * Produces an HTML response with a Content-Type of text/plain and a default
```
Comment (akrabat) [inline src/Response/TextResponse.php]: Not HTML.

**Comment by mnapoli** [comment]
Thanks @akrabat I've updated the commit.

**Comment by weierophinney** [comment]
> I did not add a charset as the HtmlResponse didn't have one. Maybe this is something to add?

Yes, please. :)

**Comment by weierophinney** [comment]
Looks good; I'll wait for your feedback as to whether or not you want to add charset to this PR as well before merging, however.

**Comment by mnapoli** [comment]
Will do tomorrow! I'll add it on HtmlResponse too. UTF-8 by default?

**Comment by weierophinney** [comment]
> Will do tomorrow! I'll add it on HtmlResponse too.

Awesome!

> UTF-8 by default?

Of course!

**Comment by mnapoli** [comment]
I've pushed a new commit (as I think this is a separate change since it also applies to HtmlResponse). 

From what I have read online (example: http://stackoverflow.com/a/19815111/245552) JSON is UTF-8 (mandatory) so there is no need for a charset.

**Comment by weierophinney** [comment]
Merged to develop for release with 1.2.0. Thanks, @mnapoli !

**Pull request closed without merging**